osu.Game/Database/StableImportManager.cs

```diff
@@ -0,0 +1,96 @@
+using osu.Framework.Platform;
+using osu.Game.Beatmaps;
+using osu.Game.Collections;
+using osu.Game.IO;
+using osu.Game.Overlays;
+using osu.Game.Overlays.Settings.Sections.Maintenance;
+using osu.Game.Scoring;
+using osu.Game.Skinning;
+
+namespace osu.Game.Database
+{
+    public class StableImportManager : Component
+    {
+        [Resolved]
+        private SkinManager skins { get; set; }
+
+        [Resolved]
+        private BeatmapManager beatmaps { get; set; }
+
+        [Resolved]
+        private ScoreManager scores { get; set; }
+
+        [Resolved]
+        private CollectionManager collections { get; set; }
+
+        [Resolved]
+        private OsuGame game { get; set; }
+
+        [Resolved]
+        private DialogOverlay dialogOverlay { get; set; }
+
+        [Resolved(CanBeNull = true)]
+        private DesktopGameHost desktopGameHost { get; set; }
+
+        private StableStorage cachedStorage;
+
+        public bool SupportsImportFromStable => RuntimeInfo.IsDesktop;
+
+        public async Task ImportFromStableAsync(StableContent content)
+        {
+            var stableStorage = await getStableStorage().ConfigureAwait(false);
+            var importTasks = new List<Task>();
+
+            Task beatmapImportTask = Task.CompletedTask;
+            if (content.HasFlagFast(StableContent.Beatmaps))
+                importTasks.Add(beatmapImportTask = beatmaps.ImportFromStableAsync(stableStorage));
+
+            if (content.HasFlagFast(StableContent.Skins))
+                importTasks.Add(skins.ImportFromStableAsync(stableStorage));
+
+            if (content.HasFlagFast(StableContent.Collections))
+                importTasks.Add(beatmapImportTask.ContinueWith(_ => collections.ImportFromStableAsync(stableStorage), TaskContinuationOptions.OnlyOnRanToCompletion));
+
+            if (content.HasFlagFast(StableContent.Scores))
+                importTasks.Add(beatmapImportTask.ContinueWith(_ => scores.ImportFromStableAsync(stableStorage), TaskContinuationOptions.OnlyOnRanToCompletion));
+
+            await Task.WhenAll(importTasks.ToArray()).ConfigureAwait(false);
+        }
+
+        private async Task<StableStorage> getStableStorage()
+        {
+            var stableStorage = game.GetStorageForStableInstall();
+            if (stableStorage != null)
+                return stableStorage;
+
+            if (cachedStorage != null)
+                return cachedStorage;
+
+            var taskCompletionSource = new TaskCompletionSource<string>(TaskCreationOptions.RunContinuationsAsynchronously);
+            Schedule(() => dialogOverlay.Push(new StableDirectoryLocationDialog(taskCompletionSource)));
+            var stablePath = await taskCompletionSource.Task.ConfigureAwait(false);
+
+            return cachedStorage = new StableStorage(stablePath, desktopGameHost);
+        }
+    }
+
+    [Flags]
+    public enum StableContent
+    {
+        Beatmaps = 1,
+        Scores = 2,
+        Skins = 4,
+        Collections = 8,
```

Comment:
It's a code style thing, but I prefer flag enums to use bit shifts explicitly instead of raw literals (so `1 << 1`, `1 << 2`, etc.)

Not going to insist on it as there is not really a clear convention in this repository (can find examples of both) but yeah.